OutOfSchool/OutOfSchool.WebApi/Controllers/WorkshopController.cs

```diff
@@ -99,14 +108,18 @@ public async Task<ActionResult> Update(WorkshopDTO workshopDto)
         /// <param name="id">Workshop's key.</param>
         /// <returns>A <see cref="Task{TResult}"/> representing the result of the asynchronous operation.</returns>
         [Authorize(Roles = "parent,admin")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
         [HttpDelete("{id}")]
-        public async Task<ActionResult> Delete(long id)
+        public async Task<IActionResult> Delete(long id)
         {
-            if (id == 0)
+            if (id < 1)
             {
-                return BadRequest("Id cannot be 0.");
+                throw new ArgumentOutOfRangeException(
```

Comment:
we need to pass parameter name to the ArgumentOutOfRangeException constructor
id.ToString() is not the parameter name

Reply:
![image](https://user-images.githubusercontent.com/55458556/110956723-97f6f300-8353-11eb-8547-ff285b89968c.png)
yes, it should be the parameter name but casted to the string